pkg/Cpanel/Security/Advisor/Assessors/Apache.pm

```diff
@@ -194,7 +192,9 @@ sub _centos_symlink_protection {
     my $local_settings = ($is_ea4) ? Cpanel::DataStore::fetch_ref('/var/cpanel/conf/apache/local') : undef;
     my $bluehost_ea4 = ($is_ea4) ? ( exists $local_settings->{main}->{symlink_protect} && $local_settings->{main}->{symlink_protect}->{item}->{symlink_protect} eq 'On' ) : 0;
 
-    if ($ruid) {
+    my $kernelcare_state = Cpanel::KernelCare::get_kernelcare_state();
+
+    if ( $ruid and $kernelcare_state != $Cpanel::KernelCare::KC_FREE_PATCH_SET and $kernelcare_state != $Cpanel::KernelCare::KC_EXTRA_PATCH_SET ) {
```

Comment:
There is a lot of duplication of this condition: `$kernelcare_state != $Cpanel::KernelCare::KC_FREE_PATCH_SET and $kernelcare_state != $Cpanel::KernelCare::KC_EXTRA_PATCH_SET`

Could we refactor that out into a function for maintainability?

Reply:
You mean put a subroutine wrapper around these constants? I'm not sure how that'd improve maintainability. Can you give me an example of what you mean?

Reply:
Sure. For example:
```perl
sub _has_kernelcare {
    my ($kernelcare_state) = @_;
   return $kernelcare_state == $Cpanel::KernelCare::KC_FREE_PATCH_SET || $kernelcare_state == $Cpanel::KernelCare::KC_EXTRA_PATCH_SET
}
...
my $has_kernelcare = _has_kernelcare($kernelcare_state);
if ($ruid && !$has_kernelcare) {
    ....
}
if (($bluehost_ea3 || $bluehost_ea4) && !$has_kernelcare) {
    ...
}
```

This makes the logic clearer, since we can see quickly what the condition is.

Reply:
Oh I see what you're saying. Sure, thank you.